### PR TITLE
WT-8985 Extend test_tiered07.py to test all WT_SESSION::drop options

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

File: test/suite/test_tiered07.py [hunk 1]
```diff
@@ -147,7 +160,11 @@ def test_tiered(self):
 
         # Create new table with new name.
         self.pr('create new table')
-        self.session.create(self.newuri, 'key_format=S')
+        self.session.create(self.newuri, 'key_format=S,value_format=S,tiered_storage=(name=none)')
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_tiered07.py:163]: Maybe we should create another table and leave `self.newuri` without any changes in config?

File: test/suite/test_tiered07.py [hunk 2]
```diff
@@ -147,7 +160,11 @@ def test_tiered(self):
...
+
+        # Check that the underlying files are not removed when configured in drop.
+        self.session.drop(self.newuri, 'remove_files=false')
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_tiered07.py:166]: It's good that we test removal of non-tiered tables, but we also should test tiered tables and check if the underlying wtobj file is deleted/not deleted.

Comment (mm-ng) [reply test/suite/test_tiered07.py:166]: My bad, I meant to create a tiered table.

Comment (mm-ng) [reply test/suite/test_tiered07.py:166]: Currently, the wtobj files are not being removed, regardless of whether we use the remove_files configuration or not.

File: test/suite/test_tiered07.py [hunk 3]
```diff
@@ -130,6 +131,23 @@ def test_tiered(self):
         self.session.drop(self.localuri)
         self.session.drop(self.uri)
 
+        # FIXME: The drop operation for tiered tables should both remove the files from the metadata
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_tiered07.py:134]: Please add `WT-9003` to the FIXME comment to link them.

File: test/suite/test_tiered07.py (hunk 2 above)
Comment (AlexeyAnisimovAU) [reply test/suite/test_tiered07.py]: Good catch! I've created a ticket for the issue: WT-9003

File: test/suite/test_tiered07.py [hunk 4]
```diff
@@ -29,11 +29,12 @@
 from helper_tiered import generate_s3_prefix, get_auth_token, get_bucket1_name
 from wtscenario import make_scenarios
 import os, wiredtiger, wttest
+from suite_subprocess import suite_subprocess
 StorageSource = wiredtiger.StorageSource  # easy access to constants
 
 # test_tiered07.py
 #    Basic tiered storage API for schema operations.
-class test_tiered07(wttest.WiredTigerTestCase):
+class test_tiered07(wttest.WiredTigerTestCase, suite_subprocess):
```
Comment (AlexeyAnisimovAU) [inline test/suite/test_tiered07.py:37]: Why do we need subprocess in this test?

Comment (mm-ng) [reply test/suite/test_tiered07.py:37]: I must have added this when I was experimenting with something but you're right, looks like we don't need it now.

**Review by AlexeyAnisimovAU — APPROVED** [review]
Well done, LGTM!

File: test/suite/test_tiered07.py [hunk 5]
```diff
@@ -130,6 +131,25 @@ def test_tiered(self):
         self.session.drop(self.localuri)
         self.session.drop(self.uri)
 
+        # By default, the remove_files configuration for drop is true. This means that the
+        # drop operation for tiered tables should both remove the files from the metadata
+        # file and remove the corresponding local object files in the directory. Currently the
+        # below code is commented as the files are not removed from the directory. FIXME: WT-9003
+        # self.assertFalse(os.path.isfile("abc-0000000001.wtobj"))
+        # self.assertFalse(os.path.isfile("abc-0000000002.wtobj"))
```
Comment (sueloverso) [inline test/suite/test_tiered07.py:139]: We should look into this as the drop call should remove the local objects. See my suggestion in WT-9003.

**Review by sueloverso — COMMENTED** [review]
These changes seem fine. I'd like investigation into why the files are not removed.

**Pull request merged**